services/QuillLMS/app/models/auth_credential.rb

```diff
@@ -24,4 +24,18 @@
 #
 class AuthCredential < ActiveRecord::Base
   belongs_to :user
+
+  def google_authorized?
+    provider == 'google' && refresh_token_valid?
+  end
+
+  def refresh_token_valid?
+    return false if expires_at.nil? || refresh_token.nil?
+
+    Time.now < refresh_token_expires_at
```

Comment:
This logic is written elsewhere slightly differently (same logic). It might be better to make both the same
```ruby
Time.now - 6.months >= current_credentials.expires_at

Reply:
Good call.  I will update